.github/workflows/generate-config.yaml

```diff
@@ -37,59 +37,78 @@ jobs:
           MERGE_COMMIT=$(git log -1 --pretty=format:"%H")
           echo "MERGE_COMMIT=$MERGE_COMMIT" >> $GITHUB_ENV
 
-      - name: Determine NETWORK
-        id: determine_network
+      - name: Determine NETWORKS
+        id: determine_networks
         run: |
           echo "Checking modified files in commit ${{ env.MERGE_COMMIT }}"
           MODIFIED_FILES=$(git diff --name-only HEAD~1..HEAD)
           echo "Modified files: $MODIFIED_FILES"
+
+          NETWORKS=()
+
           if echo "$MODIFIED_FILES" | grep -q "registry/mainnet/interchain/squid.tokenlist.json"; then
-            echo "NETWORK=mainnet" >> $GITHUB_ENV
-          elif echo "$MODIFIED_FILES" | grep -q "registry/testnet/interchain/squid.tokenlist.json"; then
-            echo "NETWORK=testnet" >> $GITHUB_ENV
-          else
+            NETWORKS+=("mainnet")
+          fi
+          if echo "$MODIFIED_FILES" | grep -q "registry/testnet/interchain/squid.tokenlist.json"; then
+            NETWORKS+=("testnet")
+          fi
+          if echo "$MODIFIED_FILES" | grep -q "registry/stagenet/interchain/squid.tokenlist.json"; then
+            NETWORKS+=("stagenet")
+          fi
+          if echo "$MODIFIED_FILES" | grep -q "registry/devnet-amplifier/interchain/squid.tokenlist.json"; then
+            NETWORKS+=("devnet-amplifier")
+          fi
+
+          if [ ${#NETWORKS[@]} -eq 0 ]; then
             echo "No relevant changes detected."
-            echo "NETWORK=none" >> $GITHUB_ENV
+            echo "NETWORKS=none" >> $GITHUB_ENV
             exit 0
           fi
 
+          echo "Detected networks: ${NETWORKS[*]}"
+          echo "NETWORKS=${NETWORKS[*]}" >> $GITHUB_ENV
       - name: Checkout external repository
-        if: env.NETWORK != 'none'
+        if: env.NETWORKS != 'none'
```

Comment:
instead of adding this check on every step below, can we do this check in a separate step and exit (1 or non zero) the run early with some type of message.

Reply:
@RiceAndMeet exiting with 1 or non-zero in the workflow means that the execution *has failed* (not an expected behavior), which shouldn't be the case if other files were edited besides the ones we are tracking and the behavior should be that other steps need to be *skipped* and in order to do that, we can either rely on the `if` conditions or brining these checks inside the `run` for `soft-skips`.

Besides this, there is a separate step that does NETWORKS checks [here](https://github.com/axelarnetwork/axelar-configs/pull/221/files/a662bbd3084c65232e5c3e08636c55d279a94887#diff-975ef3e32000879ef56e4a2f2e382e7f110744dfaa364d6a6235159d1e9463d7R65).

Let me know if you want to discuss this further async.